### DISCUSSION TITLE
Update of RollingFile & OnStartupTriggeringPolicy documentation would be nice

### DISCUSSION THREAD

**Discussion by michal-tuma** [apache/logging-log4j2#2675] [Ideas]
Hello,
I spend couple of hours understanding why my log4j config don't rollover the log file on startup (OnStartupTriggeringPolcy) using my configured composite policy:

```
      <Policies>
        <OnStartupTriggeringPolicy/>
        <SizeBasedTriggeringPolicy size="1 GB"/>
        <TimeBasedTriggeringPolicy/>
      </Policies>
```

I found out that I'm just super stupid. Like many-times before. I had defined `append=false` in RollingFileAppender.
That is the reason why OnStartupTrigerringPolicy did not rollover log file when JVM starts:
 1. JVM starts
 2. rolling file appender (because have append = false) create new active log file (empty one)
 3. startup trigerring policy check my active log file & because it has size = 0 it just do not perform rollover
 THAT'S IT, WORKS AS CODED, I'M STUPID.
 
 But that takes me 3 hours to understand.
 
 Please can you just update the https://logging.apache.org/log4j/2.x/manual/appenders.html#onstartup-triggering-policy with a note in a sence: 
 WARNING: Be sure you don't set `append="false"` for a rolling file appender in a combination with OnStartupTrigerringPolicy. Otherwise before OnStartupTrigerringPolicy is being evaluated the rolling file appender create empty file. Thus the old log file is deleted before policy is evaluated. Which finally ends up the old file is not rollovered at all.
 
 Or similar, more explainable warning. Just to save peoples time in future before they realize the same as me.
 Thank you very much in advance,
 Michal

**Comment by ppkarwasz** [comment]
@michal-tuma,

Thanks, we'll take this into account in #2528.

BTW: why did you choose `append="false"`? It is not the default mode and it requires an additional counter to keep track of the current position.

**Reply by michal-tuma** [reply]
It was a mistake. I was suggesting it is the same as `additivity="false"` on loggers. Then I just forget about that. 
But for errors we have to pay - I spend couple of hours to understand, that was my price.
:)
